test_manager/features/steps/tck_step_implementations.py

```diff
@@ -25,34 +25,19 @@
 # -------------------------------------------------------------------------
 import base64
 import codecs
-import time
+from typing import Any, Dict
+from uprotocol.proto.ustatus_pb2 import UCode
 
 from behave import when, then, given
 from behave.runner import Context
 from hamcrest import assert_that, equal_to
 
 
-@given(u'"{sdk_name}" creates data for "{command}"')
-@when(u'"{sdk_name}" creates data for "{command}"')
+@given('"{sdk_name}" creates data for "{command}"')
+@when('"{sdk_name}" creates data for "{command}"')
 def create_sdk_data(context, sdk_name: str, command: str):
-    context.logger.info("Inside create register listener data")
     context.json_dict = {}
     context.status_json = None
-    if command == "send":
-        context.on_receive_msg.pop(sdk_name, None)
-    elif command == "invokemethod":
-        context.on_receive_rpc_response.pop(sdk_name, None)
-    elif command == "uri_serialize":
-        context.on_receive_serialized_uri = None
-    elif command == "uri_deserialize":
-        context.on_receive_deserialized_uri = None
-    elif command == "uri_validate":
-        context.on_receive_validation_result.pop(sdk_name, None)
-        context.on_receive_validation_msg.pop(sdk_name, None)
-    elif command == "uuid_serialize":
-        context.on_receive_serialized_uuid = None
-    elif command == "uuid_deserialize":
```

Comment:
This I understood: we're utilizing the common variable context.response_data across all commands. Where do we reset or clear the previous values stored in it?

Reply:
We do not erase values at this point in that variable. Instead, we are tracking responses based on "test id"s. In this way, we don't have a concern about reading old values, since first we check if the response matches to the request we sent. This does raise a question about if a response is never sent, which we will fix in this PR.